### PR TITLE
Add run-frontier-squid-container doc (SOFTWARE-4268)

### PR DESCRIPTION
**Pull request by edquist**
Cobbled together from
- run-stashcache-container.md
- frontier-squid.md
- docker-frontier-squid/README.md

### CODE REVIEW

File: docs/data/run-frontier-squid-container.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+
+Before Starting
+---------------
+
+Before starting the installation process, consider the following points (consulting [the Reference section below](#reference) as needed):
+
+1. **Docker:** For the purpose of this guide, the host must have a running docker service
+   and you must have the ability to start containers (i.e., belong to the `docker` Unix group).
+-   **User IDs:** If it does not exist already, the installation will create the `squid` Linux user
```
Comment (brianhlin) [inline docs/data/run-frontier-squid-container.md:35]: Since it's in a container I don't think we really care about the UID

File: docs/data/run-frontier-squid-container.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+-   **Network ports:** Frontier squid communicates on ports 3128 (TCP) and 3401 (UDP)
```
Comment (brianhlin) [inline docs/data/run-frontier-squid-container.md:36]:
```suggestion
1.   **Network ports:** Frontier squid communicates on ports 3128 (TCP) and 3401 (UDP)
```

File: docs/data/run-frontier-squid-container.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+-   **Host choice:** If you will be supporting the Frontier application at your site, review the
```
Comment (brianhlin) [inline docs/data/run-frontier-squid-container.md:37]:
```suggestion
1.   **Host choice:** If you will be supporting the Frontier application at your site, review the
```

@efajardo @sfiligoi do you guys bother with PVCs for squid caches?

File: docs/data/run-frontier-squid-container.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+### Environment variables ###
+
+In addition to the required configuration above (ports and file systems),
+you may also configure the behavior of your cache with the following variables using an environment variable file:
+
+Where the environment file on the docker host, `/etc/squid/.env`, has any of the following variables set in `KEY=VALUE`
+format:
+
+Variable name       | Description                                                             | Defaults                                     |
```
Comment (brianhlin) [inline docs/data/run-frontier-squid-container.md:52]: Do these tables work without a leading `|`?

File: docs/data/run-frontier-squid-container.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+-   **User IDs:** If it does not exist already, the installation will create the `squid` Linux user
+-   **Network ports:** Frontier squid communicates on ports 3128 (TCP) and 3401 (UDP)
+-   **Host choice:** If you will be supporting the Frontier application at your site, review the
+[upstream documentation](https://twiki.cern.ch/twiki/bin/view/Frontier/InstallSquid#Hardware) to determine how to size your equipment.
+
+
+Configuring Squid
+-----------------
+
+### Environment variables ###
```
Comment (brianhlin) [inline docs/data/run-frontier-squid-container.md:44]:
```suggestion
### Environment variables (optional) ###
```

File: docs/site-planning.md [hunk 6]
```diff
@@ -82,8 +82,8 @@ OSG supports the use of the [Singularity](http://singularity.lbl.gov/) container
 
 Whether accessed through CVMFS or command-line software like `curl`, the majority of software is moved via HTTP in
 cache-friendly patterns.
-All sites are highly encouraged to use an [HTTP proxy](/data/frontier-squid) to reduce the load on the WAN from the
-cluster.
+All sites are highly encouraged to use an [HTTP proxy](/data/run-frontier-squid-container) to reduce the load on the
```
Comment (brianhlin) [inline docs/site-planning.md:85]: For the time being, let's keep this linked to the RPM install

**Review by brianhlin — COMMENTED** [review]
Looks pretty good. A few comments and you'll need to add to list the file in `mkdocs.yml`

File: docs/data/run-frontier-squid-container.md (hunk 3 above)
Comment (efajardo) [reply docs/data/run-frontier-squid-container.md:37]: Yes I do. But I guess for a squid you can go either way.

**Comment by DrDaveD** [comment]
Can we move forward with this?

**Comment by brianhlin** [comment]
Hi @DrDaveD it's on the todo list. I'll bump it in prio.

File: docs/data/run-frontier-squid-container.md (hunk 4 above)
Comment (edquist) [reply docs/data/run-frontier-squid-container.md:52]: as far as i can tell yes

https://github.com/opensciencegrid/docs/blob/940ad5ee2d77b0701ed12267f3f53b494add63eb/docs/data/run-frontier-squid-container.md

File: docs/data/run-frontier-squid-container.md [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+over regular squid for common grid applications, especially Frontier
+and CVMFS. The OSG distribution of frontier-squid is a straight rebuild of the
+upstream frontier-squid package for the convenience of OSG users.
+
+This document outlines how to run Frontier Squid in a Docker container.
+
+## Frontier Squid Is Recommended
+
+OSG recommends that all sites run a caching proxy for HTTP and HTTPS
+to help reduce bandwidth and improve throughput. To that end, Compute
```
[the remark this answers is not in the document]
Comment (edquist) [reply docs/data/run-frontier-squid-container.md]: ok!

File: docs/data/run-frontier-squid-container.md [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+
+```console
+user@host $ export http_proxy=http://localhost:<HOST PORT>
+user@host $ wget -qdO/dev/null http://frontier.cern.ch 2>&1 | grep X-Cache
+X-Cache: MISS from 797a56e426cf
+user@host $ wget -qdO/dev/null http://frontier.cern.ch 2>&1 | grep X-Cache
+X-Cache: HIT from 797a56e426cf
+```
+
+## Registering Frontier Squid
```
[the remark this answers is not in the document]
Comment (edquist) [reply docs/data/run-frontier-squid-container.md:124]: Done i think; let me know if you had in mind something different

File: docs/data/run-frontier-squid-container.md (hunk 1 above)
Comment (edquist) [reply docs/data/run-frontier-squid-container.md]: dropped the User IDs line

File: docs/site-planning.md (hunk 6 above)
Comment (edquist) [reply docs/site-planning.md]: ok!

**Comment by edquist** [comment]
@brianhlin, i think i've added your suggestions, let me know if i missed anything!

File: docs/data/run-frontier-squid-container.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+
+Before Starting
+---------------
+
+Before starting the installation process, consider the following points (consulting [the Reference section below](#reference) as needed):
+
+1. **Docker:** For the purpose of this guide, the host must have a running docker service
+   and you must have the ability to start containers (i.e., belong to the `docker` Unix group).
+1.   **Network ports:** Frontier squid communicates on ports 3128 (TCP) and 3401 (UDP)
```
Comment (brianhlin) [inline docs/data/run-frontier-squid-container.md:26]: While we're at it, let's expand on this and the same line in https://opensciencegrid.org/docs/data/frontier-squid/ using guidance from Dave's response here https://support.opensciencegrid.org/public/tickets/fc3a0dc0ab486c69fbf40e976eab6d54ecad10970785635586321e56c2c4c579

File: docs/data/run-frontier-squid-container.md [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+    OSG recommends that all sites run a caching proxy for HTTP and HTTPS to help reduce bandwidth and improve
+    throughput.
+
+This document outlines how to run Frontier Squid in a Docker container.
+
+
+Before Starting
+---------------
+
+Before starting the installation process, consider the following points (consulting [the Reference section below](#reference) as needed):
```
Comment (brianhlin) [inline docs/data/run-frontier-squid-container.md:22]: There's nor reference section in this doc, which is causing the CI to fail: https://github.com/opensciencegrid/docs/pull/749/checks?check_run_id=1736329950#step:6:14

**Review by brianhlin — CHANGES_REQUESTED** [review]
Looking pretty good. Two more comments.

File: docs/data/run-frontier-squid-container.md (hunk 10 above)
Comment (edquist) [reply docs/data/run-frontier-squid-container.md]: dropping the reference to the non-existent section

File: docs/data/run-frontier-squid-container.md (hunk 9 above)
Comment (edquist) [reply docs/data/run-frontier-squid-container.md:26]: @brianhlin, i suspect you are referring to this comment from Dave:

>Next, generally people do not filter the outgoing connections from squid, but cvmfs always uses ports 8000, 80, or 8080.   If in the future you allow CMS or ATLAS to run jobs at your site, they will use the Frontier application which also normally uses port 8000, although ATLAS does have some odd ports used by some of their Frontier servers.
>
>Finally, we really encourage sites to allow monitoring on port 3401 via UDP from CERN.  The OSG frontier-squid container does support that.  Usually it also requires some enabling from iptables and/or site firewalls.  If it cannot be done it is not a showstopper, but it really helps with diagnosing problems.  The details are at
>    https://twiki.cern.ch/twiki/bin/view/Frontier/InstallSquid#Enabling_monitoring

What specifically would you like to incorporate into the Before Starting section?  Just the 80/8000/8080 port numbers?

File: docs/data/run-frontier-squid-container.md (hunk 10 above)
Comment (brianhlin) [reply docs/data/run-frontier-squid-container.md]: I'd actually prefer that we link to this reference section: https://opensciencegrid.org/docs/data/frontier-squid/#reference. Seems relevant.

**Comment by brianhlin** [comment]
I think CI is failing because you need a corresponding `mkdocs.yml` entry

File: docs/data/run-frontier-squid-container.md (hunk 9 above)
Comment (edquist) [reply docs/data/run-frontier-squid-container.md]: Alright updated, hopefully to better answer Dan's questions

File: docs/data/run-frontier-squid-container.md [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+---------------
+
+Before starting the installation process, consider the following points (consulting [the Frontier Squid Reference section](/data/frontier-squid/#reference) as needed)::
+
+1. **Docker:** For the purpose of this guide, the host must have a running docker service
+   and you must have the ability to start containers (i.e., belong to the `docker` Unix group).
+1.   **Network ports:** Frontier squid communicates on ports 3128 (TCP) and 3401 (UDP).
+     - We encourage sites to allow monitoring on port 3401 via UDP from CERN IP address ranges, 128.142.0.0/16,
+       188.184.128.0/17, 188.185.48.0/20 and 188.185.128.0/17.
+     - If utgoing connections are filtered, note that CVMFS always uses ports 8000, 80, or 8080.
```
Comment (brianhlin) [inline docs/data/run-frontier-squid-container.md:29]:
```suggestion
     - We encourage sites to allow monitoring on port 3401 via UDP from CERN IP address ranges, 128.142.0.0/16,
       188.184.128.0/17, 188.185.48.0/20 and 188.185.128.0/17.
     - If outgoing connections are filtered, note that CVMFS always uses ports 8000, 80, or 8080.
```

In addition to the typo fix above, let's link to the CERN doc for IP ranges instead of listing the ranges. That way we don't have to track any changes to the IP ranges ourselves.

**Review by brianhlin — APPROVED** [review]
One minor comment but otherwise LGTM. Pre-approving

File: docs/data/run-frontier-squid-container.md [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+Before Starting
+---------------
+
+Before starting the installation process, consider the following points (consulting [the Frontier Squid Reference section](/data/frontier-squid/#reference) as needed)::
+
+1. **Docker:** For the purpose of this guide, the host must have a running docker service
+   and you must have the ability to start containers (i.e., belong to the `docker` Unix group).
+1.   **Network ports:** Frontier squid communicates on ports 3128 (TCP) and 3401 (UDP).
+     - We encourage sites to allow monitoring on port 3401 via UDP from CERN IP address ranges, 128.142.0.0/16,
+       188.184.128.0/17, 188.185.48.0/20 and 188.185.128.0/17.
```
Comment (edquist) [inline docs/data/run-frontier-squid-container.md:28]:
```suggestion
       188.184.128.0/17, 188.185.48.0/20 and 188.185.128.0/17.
       See the
       [CERN monitoring documentation](https://twiki.cern.ch/twiki/bin/view/Frontier/InstallSquid#Enabling_monitoring)
       for additional details.
```

**Comment by edquist** [comment]
Alright, updated!

File: mkdocs.yml [hunk 13]
```diff
@@ -39,6 +39,7 @@ nav:
 - Data:
   - HTTP & OASIS:
     - 'Install Squid HTTP Cache': 'data/frontier-squid.md'
+    - 'Running Frontier Squid in a Container': 'data/run-frontier-squid-container.md'
```
Comment (brianhlin) [inline mkdocs.yml:42]:
```suggestion
    - 'Running Frontier Squid in a Container': 'data/run-frontier-squid-container.md'
    - 'Install Frontier Squid RPM': 'data/frontier-squid.md'
```

I'd like to push folks towards the container so I'm putting this first. Adding as a suggestion since I'm too lazy to make a separate PR :)

**Comment by edquist** [comment]
@brianhlin, are these errors just because this is a new doc?  (That is, because it does not yet exist in the opensciencegrid fork / is not yet published under opensciencegrid.org?)

```

- ./site/data/run-frontier-squid-container/index.html
  *  External link https://github.com/opensciencegrid/docs/edit/master/docs/data/run-frontier-squid-container.md failed: 404 No error
  *  External link https://opensciencegrid.org/docs/data/run-frontier-squid-container/ failed: 404 No error
  * ```

**Comment by brianhlin** [comment]
> @brianhlin, are these errors just because this is a new doc? (That is, because it does not yet exist in the opensciencegrid fork / is not yet published under opensciencegrid.org?)

Yup, I never was able to get to the bottom of it in https://github.com/opensciencegrid/technology/pull/855 :(. Overriding CI and merging
